### PR TITLE
ZCS-9159 : Adding zimbraPasswordBlockCommonEnabled pref related UI.

### PR DESCRIPTION
**Pull request by BhavinBathani**
Adding facility to allow user to block common password.
This facility is now available to user and COS.

### CODE REVIEW

File: WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js [hunk 1]
```diff
@@ -3287,7 +3288,8 @@ textFieldCssClass:"admin_xform_number_input"}
                         			ZaAccount.A_zimbraPasswordMinDigitsOrPuncs,
                         			ZaAccount.A_zimbraMinPwdAge,
                         			ZaAccount.A_zimbraMaxPwdAge,
-                        			ZaAccount.A_zimbraEnforcePwdHistory]]],
+                                    ZaAccount.A_zimbraEnforcePwdHistory,
```
Comment (silentsakky) [inline WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js:3291]: check indentation in entire PR

File: WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js [hunk 2]
```diff
@@ -3372,6 +3374,12 @@ textFieldCssClass:"admin_xform_number_input"}
 									txtBoxLabel:ZaMsg.LBL_zimbraEnforcePwdHistory, labelLocation:_LEFT_, 
 									textFieldCssClass:"admin_xform_number_input",
 									visibilityChecks:[],enableDisableChecks:[[ZaAccountXFormView.isAuthfromInternalSync, entry.name, ZaAccount.A_name]]
+								},{ref:ZaAccount.A_zimbraPasswordBlockCommonEnabled, type:_SUPER_CHECKBOX_, 
+									resetToSuperLabel:ZaMsg.NAD_ResetToCOS, 
+									msgName:ZaMsg.NAD_RejectCommonPwd,
+									checkBoxLabel:ZaMsg.NAD_RejectCommonPwd, 
+								 	trueValue:"TRUE", falseValue:"FALSE",
+									visibilityChecks:[],enableDisableChecks:[[ZaAccountXFormView.isAuthfromInternalSync, entry.name, ZaAccount.A_name]]
```
Comment (silentsakky) [inline WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js:3382]: what does isAuthfromInternalSync check do?

File: WebRoot/js/zimbraAdmin/cos/model/ZaCos.js [hunk 3]
```diff
@@ -637,6 +638,7 @@ ZaCos.myXModel = {
         {id:ZaCos.A_zimbraMaxPwdAge, type:_NUMBER_, ref:"attrs/"+ZaCos.A_zimbraMaxPwdAge, maxInclusive:2147483647, minInclusive:0},
         {id:ZaCos.A_zimbraEnforcePwdHistory, type:_NUMBER_, ref:"attrs/"+ZaCos.A_zimbraEnforcePwdHistory, maxInclusive:2147483647, minInclusive:0},
         {id:ZaCos.A_zimbraPasswordLocked, type:_ENUM_, choices:ZaModel.BOOLEAN_CHOICES, ref:"attrs/"+ZaCos.A_zimbraPasswordLocked},
+        {id:ZaCos.A_zimbraPasswordBlockCommonEnabled, type:_ENUM_, choices:ZaModel.BOOLEAN_CHOICES, ref:"attrs/"+ZaCos.A_zimbraPasswordBlockCommonEnabled},
```
Comment (silentsakky) [inline WebRoot/js/zimbraAdmin/cos/model/ZaCos.js:641]: should this be _COS_ENUM_ as per https://github.com/Zimbra/zm-admin-console/pull/70/files#diff-6f3991f8de85af7e3fb7f112d8494239R1894 ?

File: WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js (hunk 1 above)
Comment (BhavinBathani) [reply WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js]: resolved but maintained parity with existing code.

File: WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js (hunk 2 above)
Comment (BhavinBathani) [reply WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js:3383]: According to my understanding by observing the helper functions, it seems a flag which represent if the authentication made by zimbra internally or by external parties.
We are keeping some flags disabled if the authentication made by external parties for preferences related to Password-management.
So i did use the same.

You can share your thoughts here, we will act accordingly.

File: WebRoot/js/zimbraAdmin/cos/model/ZaCos.js (hunk 3 above)
Comment (BhavinBathani) [reply WebRoot/js/zimbraAdmin/cos/model/ZaCos.js:641]: I just take reference from another such pref “A_zimbraPasswordLocked”.
I failed to understand the underlying logic for the same.
I am open for suggestion here.

**Pull request merged**